src/databricks/labs/ucx/hive_metastore/grants.py

```diff
@@ -403,14 +405,27 @@ def _get_cluster_to_instance_profile_mapping(self) -> dict[str, str]:
 
         return cluster_instance_profiles
 
-    def get_eligible_locations_principals(self) -> dict[str, dict]:
-        cluster_locations = {}
+    def _update_warehouse_to_instance_profile_mapping(
+        self,
+    ) -> dict[str, str]:
+        warehouse_instance_profiles = {}
+        sql_config = self._ws.warehouses.get_workspace_warehouse_config()
+        if sql_config.instance_profile_arn is not None:
+            role_name = sql_config.instance_profile_arn
+            for warehouse in self._ws.warehouses.list():
+                if warehouse.id is not None:
+                    warehouse_instance_profiles[warehouse.id] = role_name
```

Comment:
can you add the integration test for AWS and attach a screenshot of it passing?

Reply:
added aws int test.

Reply:
could not test azure getting an issue when trying to set SQL data access configuration 
https://github.com/databricks/databricks-sdk-py/issues/305